content/articles/email-validation-in-flutter/index.md

```diff
@@ -0,0 +1,181 @@
+	return new MaterialApp(
+	home: new Scaffold(
+		appBar: new AppBar(
+			title: new Text("Email Verifier")),
+		),
+		floatingActionButton: const FloatingActionButton(
+			onPressed: (){//add materialpageroute of the form file.
+			},
+		child: new Icon(Icons.add),
+		),
+	);
+ }
+}
+```
+### Step4. Creating Simple Form to Contain Textfields and Buttons.
+We are going to create a simple form inside our main screen. We will have an email, password, and Login button. This page alone can show the purpose email validator without the use of the previous page above. All it does is, It collects the users data then the data is used by the email validator to verify if its valid or invalid.  
+```dart
+import 'package:flutter/material.dart';
+import 'package:email_validator/email_validator.dart';
+void main(){
+ runApp(new MyApp());
+}
+class MyApp extends StatelessWidget{
+Widget build(BuildContext context) {
+	return new MaterialApp(
+	home: new Scaffold(
+	appBar: new AppBar(
+	title: new Text("Email Verifier")),
+	floatingActionButton: const FloatingActionButton(
+		onPressed: (){},
+	child: new Icon(Icons.add),
+	),
+	body:new Form( padding: const EdgeInsets.all(16.0),
+	child:[
+		new Column(
+		children:[ new Flexible(
+			child: new Text("Email"),
+		),
+		new Flexible(child:
+		new TextField(),
+		obscureText:true,
+		decoration: new InputDecoration.collapsed( hintText: "Email",
+	),
+),
+		new Flexible(child:
+
+		new Text("Password"),), ], ), ));
+
+	}
+}
+```
+### Step5. Adding Validators.
+Now let's add validation for email and password. This is the most important part. The Validate email method checks whether the data entered is a valid email or invalid email. If the email is valid it will add the user if it is invalid it will prompt the user to enter a valid email address. This is through a pop up error message for the  invalid  email address. 
+```dart
+import 'package:flutter/material.dart';
+import 'package:email_validator/email_validator.dart';
+void main(){
+ runApp(new MyApp());
+}
+class MyApp extends StatelessWidget{
+Widget build(BuildContext context) {
+	return new MaterialApp(
+	home: new Scaffold(
+		appBar: new AppBar(
+			title: new Text("Email Verifier")),
+			floatingActionButton: const FloatingActionButton(
+			onPressed: (){},
+			child: new Icon(Icons.add),
+),
+		body:new Form( padding: const EdgeInsets.all(16.0),
+		child:[ new Column(
+			children:[
+				new Flexible(child: new Text("Email"),), validateEmail(),
+			],
+			),
+		)
+	);
+}
+
+validateEmail() async {
+	var emailRegex = await(new RegExp( r"^[_\.0-9a-zA-Z]+(\.[_\.0-9a-zA-Z]+)*@([0-9a-zA-Z][-.:\w]*(\.[a-zA\-Z]{2,}){1,})$", caseSensitive: false ));
+	List listEmail=await Validator.email(emailRegex);
+	if (listEmail.length > 0) { 
+		return new Text("Valid email");
+} 
+	else { 
+		return new Text("Invalid email address");
+	}
+}
+```
+Don't forget to click on save button.
+Your app is ready to validate your email address.
+### Conclusion
```

Comment:
Include any next or final steps summarizing the tasks the audience just completed with possible next steps.